munch/__init__.py

```diff
@@ -397,13 +397,41 @@ def munchify(x, factory=Munch):
 
         nb. As dicts are not hashable, they cannot be nested in sets/frozensets.
     """
+    return cycle_munchify(x, factory, dict())
```

Comment:
On second glance I'm not sure we should prefix everything with `cycle_`... Given this will become the default behavior, we're better off just fixing munchify...

Reply:
To clarify, I already did change `munchify`/`unmunchify` to just call the `cycle_` variants, so I did "just fix" them already. However, you can't implement this code without having `munchify` call through to `cycle_munchify`, because you need a place to create the `seen` dict.

Addressing your other comments, I'd be happy to improve naming of the 6 helper functions, but (due to the way they recursively call each other), I don't think the code would improve in clarity or correctness if we try to eliminate them. That said, depending on how you stylistically feel about functions inside functions, I can make all of this less sprawly by using them. 

I'll drop a commit to that effect in a few minutes so you can evaluate what that would actually look like.

Reply:
We can leave the `cycle_` prefix, but please make these module-private (see other comment I made)

Reply:
Have a look at the latest changes I pushed; the helper functions for breaking cycles are not at module level anymore, so the issue of being module-private is moot.

Reply:
Gotcha